types/fh-sync.d.ts

```diff
@@ -0,0 +1,121 @@
+    function globalHandleCreate(onCreate: (dataset_id: string, params: any, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function handleRead(dataset_id: string, onRead: (dataset_id: string, uid: any, meta_data: any, callback: StandardCb<any>) => void): void;
+    function globalHandleRead(onRead: (dataset_id: string, uid: string, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function handleUpdate(dataset_id: string, onUpdate: (dataset_id: string, uid: string, data: any, meta_data: any, callback: StandardCb<any>) => void): void;
+    function globalHandleUpdate(onCreate: (dataset_id: string, uid: string, data: any, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function handleDelete(dataset_id: string, onCreate: (dataset_id: string, uid: string, meta_data: any, callback: StandardCb<any>) => void): void;
+    function globalHandleDelete(onCreate: (dataset_id: string, uid: string, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function handleCollision(dataset_id: string, onCollision: (dataset_id: string, hash: string, timestamp: any, uid: string, pre: any, post: any, meta_data: any, callback: StandardCb<any>) => void): void;
+    function globalHandleCollision(onCollision: (dataset_id: string, hash: string, timestamp: Date, uid: string, pre: any, post: any, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function listCollisions(dataset_id: string, onList: (dataset_id: string, meta_data: any, callback: StandardCb<{ [hash: string]: any }>) => void): void;
+    function globalListCollisions(onList: (dataset_id: string, meta_data: any, callback: StandardCb<{ [hash: string]: any }>) => void): void;
+
+    function removeCollision(dataset_id: string, onRemove: (dataset_id: string, collision_hash: string, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function interceptRequest(dataset_id: string, onIntercept: (dataset_id: string, interceptorParams: SyncInterceptParams, callback: NoRespCb) => void): void;
+    function interceptResponse(dataset_id: string, onIntercept: (datasetId: string, interceptorParams: SyncInterceptParams, callback: NoRespCb) => void): void;
+
+    function setConfig(config: any): void;
+
+    function globalInterceptRequest(onIntercept: (dataset_id: string, interceptorParams: SyncInterceptParams, callback: NoRespCb) => void): void;
+    function globalInterceptResponse(onIntercept: (datasetId: string, interceptorParams: SyncInterceptParams, callback: NoRespCb) => void): void;
+
+    function setGlobalHashFn(datasetId: string, hashFunction: (target: any) => string): void;
+    function setRecordHashFn(datasetId: string, hashFunction: (target: string[]) => string): void;
+
+    // Legacy function please do not use.
+    function toJSON(dataset_id: string, returnData: any, cb: (err: Error, data: any) => void): void;
```

Comment:
This `toJSON` function does nothing meaningful, I can't see it defined in the customer portal docs either. I think we should not include this and maybe even look at removing it entirely if it's not documented anywhere. WDYT?